Content.Server/Botany/Systems/PlantHolderSystem.cs

```diff
@@ -261,6 +262,15 @@ private void OnInteractUsing(Entity<PlantHolderComponent> entity, ref InteractUs
             return;
         }
 
+        //Only play splosh sound when using containers that would be "dumped" into the basin (as opposed to ie. sprayed on)
+        if (TryComp(args.Used, out SpillableComponent? spill) && _solutionContainerSystem.TryGetRefillableSolution(args.Used, out var solCpmp, out var sol))
+        {
+            if(component.WateringSound != null && sol.Volume > 0)
+            {
+                _audio.PlayPvs(component.WateringSound, uid);
+            }
+        }
```

Comment:
this feels like a weird place for this logic ? none of the plantholder stuff actually handles the refilling, so this is basically just hoping that the code path of this megafunction eventually hits this and plays the sound

would make more sense imo for it to be on the refillable itself (or whatever handles this behavior i dont remember)

Reply:
I completely agree with that, I looked around the solution container code and asked on the Discord if there was any event for that, but it doesn't seem to exist for now. I'd like to implement that myself at some point but I'm just not well versed with the codebase & engine enough for that yet.
 
The code that handles spraying (directly above mine) also plays their sound here (and also handles the reagent transfer, which should probably happen elsewhere too), so I reckoned it should be fine until someone can come in and code it proper

This feels kinda hacky but seems to work, my main worry was the solution in the container being emptied before my code ran, but it hasn't been an issue in my testing.

Reply:
Nevermind, looking harder I think I found that the event I was looking for does indeed exist, gonna work on it now.

Reply:
There, just pushed code that properly uses events instead of that hacky solution

Reply:
The event I was using earlier was the solution transfer attempt, which is non ideal and might cause issues in the future.

It turns out that implementing an event for when the solution is successfully transferred wasn't nearly as hard as I thought it was gonna be, so now things are properly implemented as they should've been from the start.